code/modules/food_and_drink/cakes.dm

```diff
@@ -24,62 +24,482 @@
+					cake.UpdateOverlays(newoverlay,"[tag]")
+					src.ClearSpecificOverlays("[src.overlay_refs[i]]")
+					staticiterator--
+					i--
+					continue
+			else if(mode == 2)
+				if("[src.overlay_refs[i]]" == layer_tag) //if it finds the identifying tag for the current layer (base,second,third) it flips the toggle and starts pulling overlays
+					toggleswitch = 1
+					var/image/buffer = src.GetOverlayImage("[src.overlay_refs[i]]")
+					var/image/slicecolor = new /image('icons/obj/foodNdrink/food_dessert.dmi',"slice-overlay")
+					if(buffer.color)
+						slicecolor.color = buffer.color
+					cake.UpdateOverlays(slicecolor,"base") //setting the base overlay of the temporary slice object
+					src.ClearSpecificOverlays(layer_tag)
+					staticiterator--
+					i--
+					continue
+				if(toggleswitch) //after setting the base layer, all subsequent overlays are registered as toppings and applied to the slice
+					var/image/buffer = src.GetOverlayImage("[src.overlay_refs[i]]")
+					var/toppingpath = replacetext("[src.overlay_refs[i]]","[replacetext]","slice")
+
+					if((toppingpath == "slice-candle") || (toppingpath == "slice-candle_lit")) //special case for candles :D
+						if(toppingpath == "slice-candle")
+							candle = 1
+						else
+							candle = 2
+							src.ClearSpecificOverlays("[src.overlay_refs[i]]")
+							staticiterator--
+							i--
+						continue
+
+					var/image/toppingimage = new /image('icons/obj/foodNdrink/food_dessert.dmi',"[toppingpath]")
+					if(buffer.color)
+						toppingimage.color = buffer.color
+					cake.UpdateOverlays(toppingimage,toppingpath)
+					src.ClearSpecificOverlays("[src.overlay_refs[i]]")
+					staticiterator--
+					i--
+			else
+				break
+
+		if(src.amount > 10)
+			src.amount -= 10
+			switch(src.clayer)
+				if(2)
+					src.amount2 = 0
+				if(3)
+					src.amount3 = 0
+			if(mode == 1)
+				cake.amount = 10
+			else
+				slices = 10
+		else if((mode == 2) && (src.clayer == 1))
+			slices = src.amount
+		else
+			qdel(cake) //this will happen if someone eats a cake without slicing it and the amount math has to recalculate past an entire layer (i.e. someone takes 11 bites)
+			user.show_text("<b>OH NO! The cake was a lie!</b>","red")
+			src.clayer--
+			src.reagents.maximum_volume -= 100
+			return
 
-	attackby(obj/item/W as obj, mob/user as mob)
-		if (istype(W, /obj/item/axe) || istype(W, /obj/item/circular_saw) || istype(W, /obj/item/kitchen/utensil/knife) || istype(W, /obj/item/scalpel) || istype(W, /obj/item/sword) || istype(W,/obj/item/saw) || istype(W,/obj/item/knife/butcher))
-			if (src.sliced == 1)
-				boutput(user, "<span class='alert'>This has already been sliced.</span>")
+		src.reagents.maximum_volume -= 100
+		if(mode == 2)
+			return list(slices,candle)
+		else
+			src.clayer--
+
+	/*_______________*/
+	/*Light stuffs :D*/
+	/*‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾*/
+	New()
+		..()
+		light = new /datum/light/point
+		light.set_brightness(0.8)
+		light.set_color(0.5, 0.3, 0)
+		light.attach(src)
```

Comment:
`/datum/light` is expensive. Consider using a medium or simple light. They are simple to use. Something like `src.add_medium_light("cake_light", list(0.5*255, 0.3*255, 0, 100)` probably.

Reply:
Actually maybe you could attach the medium / simple light component instead as it handles picking up and dropping too.

Reply:
beep boop!